### ISSUE TITLE
Some translations aren't available when you pick some other translations

### ISSUE THREAD

**Issue by JoannaFalkowska** [ncase/trust#50]
I realize that's a problem with each individual fork. But isn't there a way though to actually use only translation files instead of forks and have the rest of the app shared, all in one repo? I mean i18n-like? It's not only the matter of language picker, but consistency in general after all. You can't force people to pull and keep their repos up-to-date forever. Maybe in the worst case one person should fork back all the language repos and maintain all the languages themself?

I have no idea how this codebase works so I'm just asking. @ncase?

**Comment by Alicecold** [comment]
But it will probably calm down with the translations after a while, and in worst case someone could still go through the original and back to whatever language they wish for.

**Comment by arenaq** [comment]
Or you can create a pull request for merging the original to their repo. They will receive an email. There should be no conflicts, so all they have to do is to click "Merge" and its done ;)

**Comment by nextghost** [comment]
It'd be best to have a JSON list of translations in this (English) repo that would be downloaded by translated versions through JavaScript and displayed instead of the hard-coded list in index.html. That way, new translators won't have to send pull requests to every single other translator out there.

**Comment by arenaq** [comment]
@nextghost and how would your newly added language got translated to all other languages? the best qualified people to do this are still the authors of all forks...

**Comment by nextghost** [comment]
@arenaq Why on earth would you translate language names? The link to German translation must always be labeled as "Deutsch" so that people who only speak German can actually find it. Or do you expect them to somehow decipher weird labels like "German", "Němčina" or "ドイツ語"?

**Comment by JoannaFalkowska** [comment]
@nextghost is completely right with their JSON solution. There is no reason to have a different repo for every language, current solution is very poorly thought through. But I guess it is way too late to change this setup now.

**Comment by ncase** [comment]
I agree. In hindsight I shoulda done it with JSON, and in my later projects like https://github.com/crowds I did do exactly that! I'm currently swamped with new projects & work, but if I have the time I'll get around to changing this to JSON, maybe

**Comment by arenaq** [comment]
@nextghost I would not, but I though this is what the discussion is all about. Cause if you do not want to translate the names then it's really simple - just don't put the language selection for translation and put it all to one place :)

**Comment by JoannaFalkowska** [comment]
@ncase That is great to hear, best of luck with your new projects : )

**Comment by jkoelling** [comment]
I agree that a list of all translations maintained in the main repo would be nice. 

However, some comments sound like they suggest to have the complete translation contained in a json. I'd like to point out that this might not work in all cases. For example, the German translation is quiet a bit longer than the English original and I had to adapt multiple UI elements to make everything fit together as nicely as in the original. The whole interface would need to contain lots of extra space in the English original or adapt dynamically based on the translation to make this work with a single code base for all translations.

**Comment by jkoelling** [comment]
btw: https://github.com/ncase/trust/issues/70 is a duplicate of this issue

**Comment by arenaq** [comment]
I agree with @jkoelling that maintaining all translations in one json would not be a good idea.

I believe in solution somewhere in the middle. Translations handled like they are now, but the list of available languages handled somewhere in one place and the main page would present this list. That way the author of new translation would be responsible only for adding his own translation, not the complete list on main page.
